src/cli.ts

```diff
@@ -0,0 +1,75 @@
+#!/usr/bin/env node
+
+import * as args from "command-line-args";
+import * as cmdusage from "command-line-usage";
+import * as fs from "fs";
+import * as path from "path";
+import {AvroToTypescriptCompiler} from "./components/AvroToTypescriptCompiler";
+import {ConsoleHelper} from "./helpers/ConsoleHelper";
+
+const cmdOptions = [
+    {
+        name: "compile",
+        alias: "c",
+        type: String,
+        typeLabel: "{underline schema-directory} {underline output-directory}",
+        description: "Compile schema directory into output directory",
+        multiple: true,
+    },
+    {
+        name: "help",
+        alias: "h",
+        description: "Print this usage guide.",
+    },
+];
+
+const usageOptions = [
+    {
+        header: "avro-to-typescript",
+        content: "Compile avro schemas to typescript classes with ease. It will output to set directory " +
+        "and append namespace to path.",
+    },
+    {
+        header: "Options",
+        optionList: cmdOptions,
+    },
+    {
+        content: "Project home: {underline https://github.com/degordian/avro-to-typescript}",
+    },
+];
+try {
+    args(cmdOptions);
+} catch (e) {
+    ConsoleHelper.break("Invalid value or option used");
+}
+
+const options = args(cmdOptions);
+const usage = cmdusage(usageOptions);
+
+if (options.compile) {
+        let src: string = options.compile[0];
+
+        if (src === undefined && fs.existsSync(src)) {
```

Comment:
I am not sure about this statement. If `src` not set and if `src path` exists, create console output `"No source directory"`?

Reply:
I am not sure either... Good catch!